appium-dotnet-driver/Appium/RecordScreen.cs

```diff
@@ -0,0 +1,158 @@
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace OpenQA.Selenium.Appium
+{
+    public enum VideoQuality
+    {
+        LOW, MEDIUM, HIGH, PHOTO
+    }
+
+    public enum VideoType
+    {
+        H264, MP4, FMP4
+    }
+
+    public class RecordScreenUploadOptions
+    {
+        /// <summary>
+        /// The RemotePath upload option is the path to the remote location,
+        /// where the resulting video from the previous screen recording should be uploaded.
+        /// The following protocols are supported: http/https (multipart), ftp.
+        /// Missing value (the default setting) means the content of the resulting
+        /// file should be encoded as Base64 and passed as the endpoint response value, but
+        /// an exception will be thrown if the generated media file is too big to
+        /// fit into the available process memory.
+        /// This option only has an effect if there is/was an active screen recording session
+        /// and forced restart is not enabled (the default setting).
+        /// </summary>
+        [JsonProperty(PropertyName = "remotePath")]
+        public string RemotePath;
+
+        /// <summary>
+        /// The name of the user for the remote authentication.
+        /// Only has an effect if both `remotePath` and `password` are set.
+        /// </summary>
+        [JsonProperty(PropertyName = "username")]
+        public string Username;
+
+        /// <summary>
+        /// The password for the remote authentication.
+        /// Only has an effect if both `remotePath` and `user` are set.
+        /// </summary>
+        [JsonProperty(PropertyName = "password")]
+        public string Password;
+
+        /// <summary>
+        /// The HTTP method name ('PUT'/'POST'). PUT method is used by default.
+        /// Only has an effect if `remotePath` is set.
+        /// </summary>
+        [JsonProperty(PropertyName = "method")]
+        public string Method;
+    }
+
+    public class RecordScreenBaseOptions : RecordScreenUploadOptions
+    {
+
+        /// <summary>
+        /// Whether to ignore the result of previous capture and start a new recording
+        /// immediately (`True` value). By default  (`False`) the endpoint will try to catch and return the result of
+        /// the previous capture if it's still available.
+        /// </summary>
+        [JsonProperty(PropertyName = "forceRestart")]
+        public bool? ForceRestart;
+
+        /// <summary>
+        /// The actual time limit of the recorded video in seconds.
+        /// The default value for both iOS and Android is 180 seconds (3 minutes).
```

Comment:
This is changed since Appium 1.9.1 and it is possible to set higher values, but it requires ffmpeg binary to be present in PATH